lib/oauth2/client.rb

```diff
@@ -113,7 +113,7 @@ def request(verb, url, opts = {})
           req.params.update(opts[:params]) if opts[:params]
           yield(req) if block_given?
         end
-      rescue Faraday::ConnectionFailed => e
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
```

Comment:
I agree.  We should create a namespaced timeout error, so that people can handle them with distinct logic.

Reply:
@stanhu will you add the new error class?

Reply:
@pboling I added a new class, but I wonder if these errors should inherit from a base class. Faraday has `ServerError` and `ClientError`.

Reply:
Yeah, that would be ideal.

Reply:
Oh, but actually, that would be much more disruptive.  With the inheritance code that already rescues the Faraday errors will just continue to work.  If we switch the base class much more code would potentially break.

Let's consider that for a 3.0.0 release, as it would be a major breaking change.